src/DynamoPackages/PackageLoader.cs

```diff
@@ -421,19 +428,46 @@ public void LoadCustomNodesAndPackages(IEnumerable<string> newPaths, IPreference
             }
         }
 
-        [Obsolete("Do not use. This method is deprecated and will be removed in Dynamo 3.0. Use LoadCustomNodesAndPackages(IEnumerable<string> newPaths, IPreferences preferences, CustomNodeManager customNodeManager) instead.")]
         /// <summary>
-        /// This method is called when custom nodes and packages need to be reloaded if there are new package paths.
+        /// This method is used when custom nodes and packages need to be loaded from new package paths 
+        /// that have been added to preference settings.
         /// </summary>
-        /// <param name="loadPackageParams"></param>
+        /// <param name="newPaths">New package paths to load custom nodes and packages from.</param>
         /// <param name="customNodeManager"></param>
-        public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
+        internal void LoadNewCustomNodesAndPackages(IEnumerable<string> newPaths, CustomNodeManager customNodeManager)
         {
-            foreach (var path in loadPackageParams.Preferences.CustomPackageFolders)
+            if(newPaths == null || !newPaths.Any()) return;
+
+            var preferences = (pathManager as PathManager).Preferences;
+            var packageDirsToScan = new List<string>();
+
+            foreach (var path in newPaths)
             {
-                customNodeManager.AddUninitializedCustomNodesInPath(path, false, false);
+                var packageDirectory = pathManager.PackagesDirectories.FirstOrDefault(x => x.StartsWith(path));
+                if (packageDirectory != null)
+                {
+                    packageDirsToScan.Add(packageDirectory);
+                }
             }
-            LoadAll(loadPackageParams);
+            LoadCustomNodesAndPackagesHelper(preferences, customNodeManager, newPaths, packageDirsToScan);
+
+        }
+
+        /// <summary>
+        /// LoadCustomNodesAndPackages can be used to load custom nodes and packages
+        /// from temporary paths that do not need to be added to preference settings. 
+        /// To load from temporary custom paths, initialize a local PreferenceSettings object 
+        /// and add the paths to its CustomPackageFolders property, then initialize a new 
+        /// LoadPackageParams with this preferences object and use as input to this method.
+        /// To load from custom paths that need to be persisted to the preferences, 
+        /// initialize a LoadPackageParams from an existing preferences object.
+        /// </summary>
+        /// <param name="loadPackageParams">LoadPackageParams initialized with local PreferenceSettings object containing custom package path.</param>
+        /// <param name="customNodeManager"></param>
+        public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
```

Comment:
hmm what about making the signature of this method just take a collection of paths, internally we can create the new preferences instance?

Reply:
Yes, that's a good suggestion but I wanted to wait until we got rid of `LoadPackageParams` entirely, which I was planning to do in 2.14. I can do that now if you feel that's better - obsolete this method and create a new public API.

Reply:
This signature was already exposed as public.
I guess the idea was to the changes to a minimum

Reply:
ahh, sorry, getting confused about what was public and what was not - and when 😉 - I think time is pretty short, so I'm fine with keeping it as is for now and focusing on the test failures.